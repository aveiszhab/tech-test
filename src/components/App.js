import React, { useState } from 'react';
import '../styles/App.css';
import Search from './Search';

function App() {
  
  const [searchResult, setSearchResult] = useState([]);

  return (
    <div className="App">
    <img
      className='nasa-logo'
      src='https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg'
      alt='nasa-logo'
      name='nasa-logo'
    />
    <Search setSearchResultList ={setSearchResult}/>
    </div>
  );
}

export default App;
