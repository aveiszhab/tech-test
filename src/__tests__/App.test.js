import React from 'react';
import { render, screen } from '@testing-library/react';
import App from '../components/App';

describe('App', () => {
  it('renders correctly to match the App snapshot', () => {
    const {asFragment} = render(<App />);

    expect(asFragment()).toMatchSnapshot()
  });

  it('renders App correctly', () => {
    render(<App />);

    const nasaImage = screen.getByRole('img', {name: /nasa/i});

    expect(nasaImage).toHaveClass('nasa-logo');
    expect(nasaImage).toHaveAttribute('src', expect.stringContaining('nasa-worm-logo'));
    expect(nasaImage).toHaveAttribute('alt', expect.stringContaining('nasa-logo'))
  });

});