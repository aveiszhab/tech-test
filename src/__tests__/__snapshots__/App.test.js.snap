// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly to match the App snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="nasa-logo"
      class="nasa-logo"
      name="nasa-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <form
      class="search-form"
    >
      <input
        class="search-input"
        placeholder="Search for an image"
        type="text"
      />
      <button
        class="submit-button"
        name="go"
        type="submit"
      >
        Go!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
